packages/fcl/src/interaction-template-utils/verify-dependency-pin-same-at-block.js

```diff
@@ -0,0 +1,106 @@
+import {generateDependencyPin} from "./generate-dependency-pin.js"
+import {invariant, block} from "@onflow/sdk"
+import {log, LEVELS} from "@onflow/util-logger"
+import {normalizeInteractionTemplate} from "./normalize/interaction-template.js"
+
+export async function verifyDependencyPinsSame(
+  {template, blockHeight, network},
+  opts = {}
+) {
+  invariant(
+    template != undefined,
+    "generateDependencyPin({ template }) -- template must be defined"
+  )
+  invariant(
+    typeof template === "object",
+    "generateDependencyPin({ template }) -- template must be an object"
+  )
+  invariant(
+    template.f_type === "InteractionTemplate",
+    "generateDependencyPin({ template }) -- template must be an InteractionTemplate"
+  )
+
+  template = normalizeInteractionTemplate(template)
+
+  invariant(
+    network != undefined,
+    "generateDependencyPin({ network }) network must be defined"
+  )
+  invariant(
+    blockHeight != undefined,
+    "generateDependencyPin({ blockHeight }) blockHeight must be defined"
+  )
+  invariant(
+    typeof blockHeight === "number",
+    "generateDependencyPin({ blockHeight }) blockHeight must be a number"
+  )
+
+  switch (template.f_version) {
+    case "1.0.0":
+      let templateDependenciesPlaceholderKeys = Object.keys(
+        template.data.dependencies
+      )
+
+      for (let templateDependencyPlaceholderKey of templateDependenciesPlaceholderKeys) {
```

Comment:
room for spelling errors abound. Maybe just using key will help on the iterator

`for (let key of templateDependenciesPlaceholderKeys) {`